pennylane_lightning/lightning_kokkos/lightning_kokkos.py

```diff
@@ -474,14 +474,17 @@ def __init__(  # pylint: disable=too-many-arguments
         else:
             self._wire_map = {w: i for i, w in enumerate(self.wires)}
 
-        self._statevector = LightningKokkosStateVector(num_wires=len(self.wires), dtype=c_dtype)
-
         self._c_dtype = c_dtype
         self._batch_obs = batch_obs
 
         # Kokkos specific options
         self._kokkos_args = kokkos_args
         self._sync = sync
+
+        self._statevector = LightningKokkosStateVector(
+            num_wires=len(self.wires), dtype=c_dtype, kokkos_args=kokkos_args, sync=sync
+        )
+
         if not LightningKokkos.kokkos_config:
             LightningKokkos.kokkos_config = _kokkos_configuration()
```

Comment:
Moved `kokkos_config` to the end to prevent error with CUDA devices

Reply:
Do you mean after the state vector initialization?

Reply:
Yes, It is necessary to create first the state vector and after making the `_kokkos_configutration()`. Otherwise, Kokkos CUDA shows an error related to the device